lib/logstash/codecs/auto_flush.rb

```diff
@@ -7,20 +7,48 @@ def initialize(mc, interval)
     @stopped = Concurrent::AtomicBoolean.new # false by default
   end
 
+  # def start
+  #   # can't start if pipeline is stopping
+  #   return self if stopped?
+  #   if pending?
+  #     @task.cancel
+  #     create_task
+  #   elsif finished?
+  #     create_task
+  #   # else the task is executing
+  #   end
+  #   self
+  # end
+
   def start
     # can't start if pipeline is stopping
     return self if stopped?
-    if pending?
-      @task.reset
-    elsif finished?
-      @task = Concurrent::ScheduledTask.execute(@interval) do
-        @mc.auto_flush()
-      end
-    # else the task is executing
+
+    if pending? && @task.cancel
+      create_task
+      return
+    end
+    # maybe we have a timing edge case
+    # where pending? was true but cancel failed
+    # because the task started running
+    if finished?
+      create_task
+      return
     end
+    # else the task is executing
+    # wait for task to complete
+    # flush could feasibly block on queue access
+    @task.value
+    create_task
     self
```

Comment:
I understand the fluent interface, I am just saying that we need some consistency, in the current PR not all returns in the current methods return `self` which will break the fluent interface depending of what happen in the code.

see:
https://github.com/logstash-plugins/logstash-codec-multiline/pull/24/files#diff-4ffb517712c74a5e9ee0fcf4800b6f0cR29
https://github.com/logstash-plugins/logstash-codec-multiline/pull/24/files#diff-4ffb517712c74a5e9ee0fcf4800b6f0cR36

only the last one return self.
https://github.com/logstash-plugins/logstash-codec-multiline/pull/24/files#diff-4ffb517712c74a5e9ee0fcf4800b6f0cR43